inc/saitypes.h

```diff
@@ -530,6 +531,64 @@ typedef struct _sai_acl_capability_t
 }sai_acl_capability_t;
 
 /**
+ * @brief Segment Routing TLV Types
+ */
+typedef enum _sai_tlv_type_t
+{
+    /** Ingress TLV */
+    SAI_TLV_TYPE_INGRESS,
+    /** Egress TLV */
+    SAI_TLV_TYPE_EGRESS,
+    /** Opaque TLV */
+    SAI_TLV_TYPE_OPAQUE,
+    /** HMAC TLV */
+    SAI_TLV_TYPE_HMAC,
+} sai_tlv_type_t;
+
+/**
+ * @brief Segment Routing HMAC TLV Format
+ */
+typedef struct _sai_hmac_t {
+    sai_uint32_t key_id;
+    sai_uint32_t hmac[8];
+} sai_hmac_t;
+
+/**
+ * @brief Segment Routing TLV entry
+ */
+typedef struct _sai_tlv_t {
+    sai_tlv_type_t tlv_type;
+    union {
+        sai_ip6_t ingress_node;
+        sai_ip6_t egress_node;
+        sai_uint32_t opaque_container[4];
+        sai_hmac_t hmac;
+    } entry;
+} sai_tlv_t;
+
+/**
+ * @brief List of Segment Routing TLV entries
+ */
+typedef struct _sai_tlv_list_t
+{
+    /** Number of tlv entries */
+    uint32_t count;
+    /** TLV list */
+    sai_tlv_t *list;
+} sai_tlv_list_t;
+
+/**
+ * @brief List of Segment Routing segment entries
+ */
+typedef struct _sai_segment_list_t
+{
+    /** Number of IPv6 Segment Route entries */
+    uint32_t count;
+    /** Segment list */
+    sai_ip6_t *list;
+} sai_segment_list_t;
+
+/**
```

Comment:
Thanks @kcudnik - I believe I have incorporated your comments as best as I understand them.  It should meta make correctly now.  Let me know if I haven't addressed anything